lib/stm32/common/adc_common_v1.c

```diff
@@ -573,44 +573,28 @@ the newly defined group.
 
 void adc_set_regular_sequence(uint32_t adc, uint8_t length, uint8_t channel[])
 {
-	uint32_t fifth6 = 0;
-	uint32_t fourth6 = 0;
-	uint32_t third6 = 0;
-	uint32_t second6 = 0;
-	uint32_t first6 = 0;
-	uint8_t i = 0;
+ 	uint32_t reg32[5] = { 0 };
+	uint8_t i;
 
 	if (length > ADC_SQR_MAX_CHANNELS_REGULAR) {
 		return;
 	}
 
-	for (i = 1; i <= length; i++) {
-		if (i <= 6) {
-			first6 |= (channel[i - 1] << ((i - 1) * 5));
-		}
-		if ((i > 6) & (i <= 12)) {
-			second6 |= (channel[i - 1] << ((i - 6 - 1) * 5));
-		}
-		if ((i > 12) & (i <= 18)) {
-			third6 |= (channel[i - 1] << ((i - 12 - 1) * 5));
-		}
-		if ((i > 18) & (i <= 24)) {
-			fourth6 |= (channel[i - 1] << ((i - 18 - 1) * 5));
-		}
-		if ((i > 24) & (i <= 28)) {
-			fifth6 |= (channel[i - 1] << ((i - 24 - 1) * 5));
-		}
-	}
+	for (i = 0; i <= length; i++) {
+		int i1 = i+1;
```

Comment:
just in general, you could  just do ``` i = 1; i < length;i++``` here, isntead of using this i1 variable

Reply:
But then I would have to write `channel[i-1]` one line below. Pick your poison. ;)